doc/user/content/sql/namespaces.md

```diff
@@ -9,16 +9,74 @@ menu:
     weight: 126
 ---
 
-Materialize mimics SQL standard's namespace hierarchy, which is:
+Namespaces are a logical way to organize Materialize objects. In
+organizations with multiple objects, they help avoid naming conflicts and make
+it easier to manage objects. Namespaces allow you to refer to objects by a fully
+qualified name like `namespace.object_name`.
+
+## Namespace hierarchy
+
+Materialize follows SQL standard's namespace hierarchy for most objects. The
+Materialize structure is:
 
 - Databases (highest level)
 - Schemas
-- Tables, views, sources
+    - Tables
+    - Views
+    - Materialized views
+    - Connections
+    - Sources
+    - Sinks
+    - Indexes
+    - Types
+    - Functions
+    - Secrets
 - Columns (lowest level)
 
-Each layer in the hierarchy can contain elements directly beneath it. For
+The root layer in the hierarchy can contain elements directly beneath it. For
 example, databases can contain schemas.
 
+Objects within the schema layer are not in a hierarchy and do not necessarily
+share a relationship with one another.
+
+## Independent objects
+
+The Materialize objects that exist outside the standard namespace hierarchy
+are:
+
+- Clusters
+- Cluster replicas
+- Roles
+
+These objects are independent of other objects and are not referenced by the
```

Comment:
This isn't necessarily true. Clusters have dependencies (e.g. you can't drop a cluster with active dataflows associated with it unless you use `CASCADE`). "Independent objects" doesn't feel like the right term here.